src/fable/Fable.Compiler/Replacements.fs

```diff
@@ -127,19 +127,39 @@ module Util =
             CoreLibCall ("Util", Some "toString", false, [arg])
             |> makeCall com i.range i.returnType
 
-    let toInt, toFloat =
-        let toNumber com (i: Fable.ApplyInfo) typ (arg: Fable.Expr) =
-            match arg.Type with
-            | Fable.String ->
-                GlobalCall ("Number", Some ("parse"+typ), false, [arg])
-                |> makeCall com i.range i.returnType
-            | _ ->
-                if typ = "Int"
-                then GlobalCall ("Math", Some "floor", false, [arg])
-                     |> makeCall com i.range i.returnType
-                else wrap i.returnType arg
-        (fun com i arg -> toNumber com i "Int" arg),
-        (fun com i arg -> toNumber com i "Float" arg)
+    let toFloat com (i: Fable.ApplyInfo) (arg: Fable.Expr) =
+        match arg.Type with
+        | Fable.String ->
+            GlobalCall ("Number", Some "parseFloat", false, [arg])
+            |> makeCall com i.range i.returnType
+        | _ ->
+            wrap i.returnType arg
+
+    let toInt com (i: Fable.ApplyInfo) (arg: Fable.Expr) =
+        let patternFor = function
+            | Fable.Number kind -> 
+                match kind with
+                | Int8 -> Some "($0 + 0x80 & 0xFF) - 0x80"
+                | UInt8 -> Some "$0 & 0xFF"
+                | Int16 -> Some "($0 + 0x8000 & 0xFFFF) - 0x8000"
+                | UInt16 -> Some "$0 & 0xFFFF"
+                | Int32 -> Some "($0 + 0x80000000 >>> 0) - 0x80000000"
+                | UInt32 -> Some "$0 >>> 0"
+                | Int64 -> Some "Math.trunc($0)" // only 53-bit (still better than nothing)
+                | UInt64 -> Some "($0 > 0) ? Math.trunc($0) : ($0 >>> 0)" // 53-bit positive, 32-bit negative
```

Comment:
Care is needed when emitting inline JS with multiple arg references (like `$0` in this case): if `$0` is a function call, a template like `($0 > 0) ? Math.trunc($0) : ($0 >>> 0)` will call the function three times (see #198). These cases should be replaced by a function in fable-core or by a [let-wrapping](https://github.com/fable-compiler/Fable/commit/bc101df191d637dbe0eae062b57aece0273dbcc7).